modules/python/test/test_norm.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+
+from itertools import product
+from functools import reduce
+
+import cv2 as cv
+import numpy as np
+
+from tests_common import NewOpenCVTests
+
+
+def norm_inf(x, y=None):
```

Comment:
It make sense to convert x/y to doubles before passing to numpy (issue can be observed with `uint8`):
```
x = x.astype(np.float64)
y = y if y is None else y.astype(np.float64)
```

(here and below, except hamming)